src/TestSuite/ConsoleIntegrationTestTrait.php

```diff
@@ -90,15 +91,25 @@ trait ConsoleIntegrationTestTrait
      *
      * @param string $command Command to run
      * @param array $input Input values to pass to an interactive shell
+     * @throws \Cake\TestSuite\Stub\MissingConsoleInputException
+     * @throws \RuntimeException
      * @return void
      */
     public function exec(string $command, array $input = []): void
     {
         $runner = $this->makeRunner();
 
-        $this->_out = new ConsoleOutput();
-        $this->_err = new ConsoleOutput();
-        $this->_in = new ConsoleInput($input);
+        if ($this->_out === null) {
+            $this->_out = new ConsoleOutput();
+        }
+        if ($this->_err === null) {
+            $this->_err = new ConsoleOutput();
+        }
+        if ($this->_in === null) {
+            $this->_in = new ConsoleInput($input);
+        } elseif ($input) {
+            throw new RuntimeException('You can use $input only if $_in property is null and will be reset.');
+        }
```

Comment:
Not sure why all of this is needed if you can use the `$input` array to stub out input requests. Are there other situations you need to use mocks?

Reply:
You can use the ->once() ->any() and ->..() calls depending on your needs.
Right now it is always the once exact order match.

But I can also understand if that is out of scope for this.

I just don't see any disadvantage of this conditional setup, as this is fully BC and side-by-side working to the existing way of doing it.

Reply:
Ok, I was just checking if there was a gap in how input stubbing worked, but it sounds like you prefer to use mocks for certain kinds of tests.